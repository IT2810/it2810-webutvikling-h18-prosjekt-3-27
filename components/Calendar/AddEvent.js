import React, {PureComponent} from "react";
import {
  Text,
  StyleSheet, TouchableOpacity
} from "react-native";
import Icon from "react-native-vector-icons/Ionicons";

class AddEvent extends PureComponent {

  handleAddAgendaPress = () => {
    if (this.props.date && this.props.onAddAgenda) {
      this.props.onAddAgenda(this.props.date);
    }
  };

  render() {
    return (
      <TouchableOpacity
        style={styles.itemAdd}
        onPress={this.handleAddAgendaPress}
      >
        <Icon style={{margin: 5}} name="ios-add-circle" color="green" size={28}/>
        <Text>Create event</Text>
      </TouchableOpacity>
    );
  }
}

export default AddEvent;

const styles = StyleSheet.create({
  itemAdd: {
    backgroundColor: "white",
    flex: 1,
    flexDirection: "row",
    alignItems: "center",
    borderRadius: 5,
    padding: 10,
    marginRight: 10,
    marginTop: 17
  }
});
