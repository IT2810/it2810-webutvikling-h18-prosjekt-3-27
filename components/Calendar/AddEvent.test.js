import React from "react";
import renderer from "react-test-renderer";
import AddEvent from "./AddEvent";

it("renders correctly", () => {
  const tree = renderer
    .create(<AddEvent/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
