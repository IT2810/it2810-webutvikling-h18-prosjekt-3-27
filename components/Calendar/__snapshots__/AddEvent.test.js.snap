// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "borderRadius": 5,
      "flex": 1,
      "flexDirection": "row",
      "marginRight": 10,
      "marginTop": 17,
      "opacity": 1,
      "padding": 10,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Create event
  </Text>
</View>
`;
